### ISSUE TITLE
Please remove the game assets from Wargus

### ISSUE THREAD

**Issue by HiPhish** [NixOS/nixpkgs#208078]
Currently the `wargus` package downloads non-free assets from the internet:

```nix
data = runCommand "warcraft2" {
    buildInputs = [ unzip bchunk p7zip ];
    meta.license = lib.licenses.unfree;
  } ''
    unzip ${dataDownload} "Warcraft.II.Tides.of.Darkness/Warcraft II - Tides of Darkness (1995)/games/WarcrafD/cd/"{WC2BTDP.img,WC2BTDP.cue}
    bchunk "Warcraft.II.Tides.of.Darkness/Warcraft II - Tides of Darkness (1995)/games/WarcrafD/cd/"{WC2BTDP.img,WC2BTDP.cue} WC2BTDP
    rm -r Warcraft.II.Tides.of.Darkness
    7z x WC2BTDP01.iso
    rm WC2BTDP*.{iso,cdr}
    cp -r DATA $out
  '';
```

While I appreciate wanting to make wargus work out of the box, this has the result of making the entire package non-free. And the file size looks like this is the entire game, not the demo, which could make the entire package illegal. I believe that if anyone downloads wargus he already knows that the engine requires assets and how to add them from a legitimate copy of the game.

**Comment by winterqt** [comment]
Please follow the template and ping the maintainers next time, thanks.

cc @astro

**Comment by ehmry** [comment]
This can be fixed by specify the fixed-output assets with `requireFile` ([example](https://git.sr.ht/~ehmry/sm64pc/tree/master/item/default.nix)).

**Comment by winterqt** [comment]
Yup, `requireFile` would work fine, though we probably want to have that point to the data directory. Which then leads us to having to take into account any possible variations of those assets, if they exist. Not sure.

**Comment by astro** [comment]
I just did a quick grep thru nixpkgs: this is by far the only archive.org reference. Good luck on your mission!

**Comment by winterqt** [comment]
> I just did a quick grep thru nixpkgs: this is by far the only archive.org reference.

Correct, it seems to be a mix of dead software sources and some other game related files.

> Good luck on your mission!

I can't tell -- was this meant to be passive aggressive, as it comes off as being? I don't understand what you mean by this.

**Comment by HiPhish** [comment]
Maybe I'm missing something unique about Nix, but why do we even need to require the game assets inside the package? I don't see other game ports doing that (OpenMW, vcmi, eDuke32, fheroes2, Arx Libertatis). Only DXX Rebirth is downloading additional music files, which again I don't know why it is doing this, but at it least it does not taint the license.

Why not just let the user drop in the assets into whatever directory the application expects. Maybe I don't want the remade music. Maybe I have a different language version of the game.

> Plus: not everything on archive.org is necessary illegal, as stated above with regards to software sources. Who was implying that?

I was implying that because Warcraft 2 is still being sold, e.g. on GOG, and I am not aware of any time when the game was given away by Blizzard. Obviously I am not a lawyer, but I would not hold out my neck on this one.

**Comment by winterqt** [comment]
> I was implying that because Warcraft 2 is still being sold, e.g. on GOG, and I am not aware of any time when the game was given away by Blizzard. Obviously I am not a lawyer, but I would not hold out my neck on this one.

I agree with this, to be clear. It seems that the maintainer doesn't, though. So...

**Comment by ehmry** [comment]
If none of us are lawyers then none of us are qualified to act here.

**Comment by astro** [comment]
I am right now waiting for the archive.org download so I can create a PR to use `requireFile` instead.

**Comment by HiPhish** [comment]
@astro Can you please explain why you are requiring the game assets in the package definition to begin with? They are not necessary to compile the engine. What if my version of the game has a different checksum because it is from a different region?

@ehmry Especially because we are not lawyers I believe it better to err on the side of caution.

**Comment by anund** [comment]
Wargus compiles in the path it will read game data from. In the current packaging this is a path in the store which makes install on first run not possible.

```
terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
  what():  filesystem error: cannot copy: Permission denied [/nix/store/hm53l9mx48j261k2c1psq0d4b2ygrqdr-wargus-3.3.1/share/games/stratagus/wargus/campaigns] [/home/<user>/.stratagus/data.Wargus/campaigns]
[1]    279020 IOT instruction (core dumped)  ../local-nixpkgs/result/bin/wargus
```

**Comment by winterqt** [comment]
We could always patch it?

**Comment by anund** [comment]
There are more details [here](https://github.com/NixOS/nixpkgs/pull/208180/files#r1059063458). The eventual runtime contents of `$out/share/games/stratagus/wargus` is actually two sets of files. One set is generated during build/install and the other is transformed game assets from `wartool -v -r <CD extract>/DATA`. Have a look at the source @winterqt, maybe either or both sets of files can be patched to come from somewhere dynamic?

**Comment by ehmry** [comment]
@HiPhish stop giving unsolicited legal advice.
